File: src/main/java/common/entities/EntityInventory.java
package common.entities;

import common.containers.ISlot;
import common.containers.Inventory;
import common.world.Direction;

public class EntityInventory extends Entity implements IEntityInventory {
    private Inventory inventory = new Inventory();

    public EntityInventory(String name, int health, int maxHealth) {
        super(name, health, maxHealth);
    }

    @Override
    public ISlot getSlot(int slot) {
        return inventory.getSlot(slot);
    }

    @Override
    public void setSlot(ISlot slot) {
        inventory.setSlot(slot);
    }

}

File: src/main/java/common/entities/EntityInventory.kt
package common.entities

import common.containers.ISlot
import common.containers.Inventory
import common.world.Direction

open class EntityInventory : Entity, IEntityInventory {
    private var inventory: Inventory? = null

    constructor(name: String, health: Int, maxHealth: Int, size: Int) : super(name, health, maxHealth) {
        inventory = Inventory(size)
    }

    constructor(name: String, health: Int, maxHealth: Int) : super(name, health, maxHealth) {
        inventory = Inventory()
    }

    override fun update() {}

    override fun getSlot(slot: Int): ISlot {
        return inventory!!.getSlot(slot)
    }

    override fun setSlot(slot: ISlot) {
        inventory!!.setSlot(slot)
    }

}
